### ISSUE TITLE
writing with indent is not work

### ISSUE THREAD

**Issue by ImJeremyHe** [tafia/quick-xml#372]
I wrote a function to prettify a one-line xml, but it failed.
```rs
use super::reader::Reader;
use super::writer::Writer;
use super::events::*;
fn to_tree(xml: &str) -> String {
    let mut reader = Reader::from_str(xml);
    let mut writer = Writer::new_with_indent(Vec::new(), 32, 4);
    loop {
        let mut buf = Vec::<u8>::new();
        match reader.read_event(&mut buf) {
            Ok(Event::Eof) => break,
            Ok(e) => {
                println!("{:?}", e);
                let _ = writer.write_event(e);
            },
            Err(_) => break,
        }
    }
    String::from_utf8(writer.into_inner()).unwrap()
}
```
I found that in the `read_until_open` function in reader, it returns `Text` event always. But the `write_event` function will always set
```rs
next_should_line_break = false;
```
I guess that's why it failed.
Maybe an empty text event should not be returned in the `read_event` function?

**Comment by dralley** [comment]
@ImJeremyHe You can use `Reader::trim_text` for this.  However, what it really does is trim all text events, including ones inside of elements, there's not a separate option for trimming only if a text element is "empty"

https://docs.rs/quick-xml/0.22.0/quick_xml/struct.Reader.html#method.trim_text

**Comment by ImJeremyHe** [comment]
@dralley Thank you for your reply. I just realized that what leads to this problem is my setting `trim_text` true.
But I need to keep the white spaces in the text element. Maybe I should make a PR to fix this.

**Comment by Mingun** [comment]
> Maybe an empty text event should not be returned in the `read_event` function?

Yeah, I've created #363 for this, but that is not simple change as I expected because of how buffer for store event content is handled. Events borrow from that buffer and its instance moved into generic function, so it is not possible just to add a loop here:
https://github.com/tafia/quick-xml/blob/d8727717fb062a2b2d31ce748959220a28da9f32/src/reader.rs#L511-L526
that would skip empty text events, because buffer moves at first iteration.

**Comment by Mingun** [comment]
Generating of empty events was fixed in #398; please comment here with additional info if it does not solve your case and I'll reopen
